stan/math/rev/core/arena_matrix.hpp

```diff
@@ -139,6 +135,264 @@ class arena_matrix : public Eigen::Map<MatrixType> {
   }
 };
 
+template <typename MatrixType>
+class arena_matrix<MatrixType, require_eigen_sparse_base_t<MatrixType>>
+    : public Eigen::Map<MatrixType> {
+ public:
+  using Scalar = value_type_t<MatrixType>;
+  using Base = Eigen::Map<MatrixType>;
+  using PlainObject = std::decay_t<MatrixType>;
+  using StorageIndex = typename PlainObject::StorageIndex;
+
+  /**
+   * Default constructor.
+   */
+  arena_matrix() : Base::Map(0, 0, 0, nullptr, nullptr, nullptr) {}
+  /**
+   * Constructor for CDC formatted data. Assumes compressed and does not own
+   * memory.
+   *
+   * Constructs a read-write Map to a sparse matrix of size rows x cols,
+   * containing nnz non-zero coefficients, stored as a sparse format as defined
+   * by the pointers outerIndexPtr, innerIndexPtr, and valuePtr. If the optional
+   * parameter innerNonZerosPtr is the null pointer, then a standard compressed
+   * format is assumed.
+   *
+   * @param rows Number of rows
+   * @param cols Number of columns
+   * @param nnz Number of nonzero items in matrix
+   * @param outerIndexPtr A pointer to the array of outer indices.
+   * @param innerIndexPtr A pointer to the array of inner indices.
+   * @param valuePtr A pointer to the array of values.
+   * @param innerNonZerosPtr A pointer to the array of the number of non zeros
+   * of the inner vectors.
+   *
+   */
+  arena_matrix(Eigen::Index rows, Eigen::Index cols, Eigen::Index nnz,
+               StorageIndex* outerIndexPtr, StorageIndex* innerIndexPtr,
+               Scalar* valuePtr, StorageIndex* innerNonZerosPtr = nullptr)
+      : Base::Map(rows, cols, nnz, outerIndexPtr, innerIndexPtr, valuePtr,
+                  innerNonZerosPtr) {}
+
+ private:
+  template <typename T, typename Integral>
+  auto* copy_vector(const T* ptr, Integral size) {
+    T* ret = ChainableStack::instance_->memalloc_.alloc_array<T>(size);
+    std::copy_n(ptr, size, ret);
+    return ret;
+  }
+
+ public:
+  /**
+   * Constructs `arena_matrix` from an `Eigen::SparseMatrix`.
+   * @param other Eigen Sparse Matrix class
+   */
+  template <typename T, require_same_t<T, PlainObject>* = nullptr>
+  arena_matrix(T&& other)  // NOLINT
```

Comment:
It doesn't look like any of these constructors uses the `innerNonZerosPtr` member from the inputs, is that intentional?

Reply:
Oh! I think I left them out because elsewhere in the Eigen docs it said it was just for compatibility with other packages. But I'll put them in just in case. No reason not to